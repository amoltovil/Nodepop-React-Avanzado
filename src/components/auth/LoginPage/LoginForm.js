import React from 'react';
import T from 'prop-types';

import Button from '../../shared/Button';
import FormField from '../../shared/FormField';
import useForm from '../../../hooks/useForm';

import './LoginForm.css';

function LoginForm({ onSubmit, isLoading }) {
  const [credentials, handleChange, handleSubmit] = useForm({
    email: '',
    password: '',
    rememberMe: false,
  });

  //   const handleUsernameChange = event => {
  //     const newCredentials = { ...credentials, username: event.target.value };
  //     setCredentials(newCredentials);
  //   };

  //   const handlePasswordChange = event => {
  //     const newCredentials = { ...credentials, password: event.target.value };
  //     setCredentials(newCredentials);
  //   };

 
  const handleFormSubmit = ev => {
    onSubmit(credentials);
  };

  //   const validatePassword = () => {
  //     return /abcd/.test(credentials.password);
  //   };

  const { email, password, rememberMe } = credentials;

  return (
    <form className="loginForm" onSubmit={handleSubmit(handleFormSubmit)}>
      <FormField
        type="text"
        name="email"
        label="Introduzca su email"
        className="loginForm-field"
        value={email}
        onChange={handleChange}
        autofocus
      />
      <FormField
        type="password"
        name="password"
        label="password"
        className="loginForm-field"
        value={password}
        onChange={handleChange}
      />
      <Button
        type="submit"
        className="loginForm-submit"
        variant="primary"
        disabled={isLoading || !email || !password}
      >
        Log in
      </Button>
      <br /> <br/>
      <FormField
        type="checkbox"
        name="rememberMe"
        label="Recordar ContraseÃ±a"
        className="loginForm-checkbox"
        checked={rememberMe}
        // onChange={handleChangeCheckbox}
        onChange={ handleChange}
      />
    </form>
  );
}

LoginForm.propTypes = {
  onSubmit: T.func.isRequired,
  isLoading: T.bool,
};

LoginForm.defaultProps = {
  isLoading: false,
};

export default LoginForm;
