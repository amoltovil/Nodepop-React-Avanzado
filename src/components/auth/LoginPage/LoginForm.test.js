import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from './LoginForm';

describe('LoginForm', () => {
    const props = {
        isLoading: false,
        onSubmit: jest.fn(),
    };
    
    const render = () => shallow(<LoginForm {...props} />);

    test('should render', () => {
        const wrapper = render();
        //console.log(wrapper.debug());
        expect(wrapper.exists()).toBe(true);
        expect(wrapper.find('form').props().className).toBe('loginForm');

    });

    test('snapshot testing', () => {
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });

    test('snapshot testing with is loading', () => {
        const wrapper = render();
        wrapper.setProps({ isLoading: true });
        expect(wrapper).toMatchSnapshot();
    });
    
    test('snapshot testing with an email and password', () => {
        // se renderizara con el botÃ³n habilitado
        const credentials = {email: 'Alicia', password: 'password', rememberMe: false}
        const wrapper = render();
        //wrapper.setProps({ isLoading: true });
        const emailField = wrapper.find('.loginForm-field').at(0);
        //console.log(userNameField.debug());
        emailField
            .props()
            .onChange({ target: { name: 'email', value: credentials.email } });
        const passwordField = wrapper.find('.loginForm-field').at(1);
        passwordField
            .props()
            .onChange({ target: { name: 'password', value: credentials.password } });

        const remembermeField = wrapper.find('.loginForm-checkbox');
            remembermeField
                .props()
                .onChange({ target: { name: 'rememberMe', value: credentials.rememberMe } });
    
        expect(wrapper).toMatchSnapshot();
      });

    test('should submit credentials', () => {
        const credentials = {email: 'Alicia', password: 'password', rememberMe: false}
        const wrapper = render();
        const emailField = wrapper.find('.loginForm-field').at(0);
        //console.log(userNameField.debug());
        emailField
            .props()
            .onChange({ target: { name: 'email', value: credentials.email } });
        const passwordField = wrapper.find('.loginForm-field').at(1);
        passwordField
            .props()
            .onChange({ target: { name: 'password', value: credentials.password } });

        const remembermeField = wrapper.find('.loginForm-checkbox');
            remembermeField
                .props()
                .onChange({ target: { name: 'rememberMe', value: credentials.rememberMe } });

        const form = wrapper.find('.loginForm');
        form.simulate('submit', { preventDefault: () => { }});

        expect(wrapper.find('.loginForm-submit').props().disabled).toBe(false);
        expect(props.onSubmit).toHaveBeenCalled();
        expect(props.onSubmit).toHaveBeenCalledWith(credentials);
    });
});