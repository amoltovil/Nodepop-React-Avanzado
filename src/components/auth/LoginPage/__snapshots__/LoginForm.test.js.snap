// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot testing 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <FormField
    autofocus={true}
    className="loginForm-field"
    label="Introduzca su email"
    name="email"
    onChange={[Function]}
    type="text"
    value=""
  />
  <FormField
    className="loginForm-field"
    label="password"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <styled.button
    className="loginForm-submit"
    disabled={true}
    type="submit"
    variant="primary"
  >
    Log in
  </styled.button>
  <br />
   
  <br />
  <FormField
    checked={false}
    className="loginForm-checkbox"
    label="Recordar Contraseña"
    name="rememberMe"
    onChange={[Function]}
    type="checkbox"
  />
</form>
`;

exports[`LoginForm snapshot testing with an email and password 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <FormField
    autofocus={true}
    className="loginForm-field"
    label="Introduzca su email"
    name="email"
    onChange={[Function]}
    type="text"
    value="Alicia"
  />
  <FormField
    className="loginForm-field"
    label="password"
    name="password"
    onChange={[Function]}
    type="password"
    value="password"
  />
  <styled.button
    className="loginForm-submit"
    disabled={false}
    type="submit"
    variant="primary"
  >
    Log in
  </styled.button>
  <br />
   
  <br />
  <FormField
    checked={false}
    className="loginForm-checkbox"
    label="Recordar Contraseña"
    name="rememberMe"
    onChange={[Function]}
    type="checkbox"
  />
</form>
`;

exports[`LoginForm snapshot testing with is loading 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <FormField
    autofocus={true}
    className="loginForm-field"
    label="Introduzca su email"
    name="email"
    onChange={[Function]}
    type="text"
    value=""
  />
  <FormField
    className="loginForm-field"
    label="password"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <styled.button
    className="loginForm-submit"
    disabled={true}
    type="submit"
    variant="primary"
  >
    Log in
  </styled.button>
  <br />
   
  <br />
  <FormField
    checked={false}
    className="loginForm-checkbox"
    label="Recordar Contraseña"
    name="rememberMe"
    onChange={[Function]}
    type="checkbox"
  />
</form>
`;
